app/controllers/providers/edit_initial_allocations_controller.rb

```diff
@@ -0,0 +1,119 @@
+module Providers
+  class EditInitialAllocationsController < ApplicationController
+    def do_you_want
+      if request.post?
+        case params[:allocation][:request_type]
+        when AllocationsView::RequestType::INITIAL
+          redirect_to number_of_places_provider_recruitment_cycle_allocation_path(
+            provider_code: allocation.accredited_body.provider_code,
+            recruitment_cycle_year: recruitment_cycle.year,
+            training_provider_code: allocation.provider.provider_code,
+            id: allocation.id,
+          )
+        when AllocationsView::RequestType::DECLINED
+          destroy
```

Comment:
The allocation is destroyed before the user has a chance to confirm?

Reply:
Ah I see - it redirects to the a page confirming the deletion ...
Worth being consistent with naming here? delete vs destroy